File: ruby/spec/lib/email_validator/regex_matcher_spec.rb
require 'email_validator/regex_matcher'

describe EmailValidator::RegexMatcher do
  subject(:matcher) { EmailValidator::RegexMatcher.new }

  context "email" do
    it "matches good email" do
      expect(matcher.match('prez@whitehouse.gov')).not_to be_nil
    end

    # Note that there is no useful error message to extract!
    it "fails to match bad email with incomplete domain" do
      expect(matcher.match('prez@whitehouse.')).to be_nil
    end
  end
end

File: scala/src/test/scala/com/franklinchen/RegexMatcherSpec.scala
package com.franklinchen

import org.specs2._
import org.specs2.specification._

trait RegexMatcherSpec extends Specification { def is = s2"""
  ${matcher.name}
    matches good email                              $e1
    fails to match bad email with incomplete domain $e2
  """

  def e1 =
    matcher.matches("prez@whitehouse.gov") must beTrue

  def e2 =
    matcher.matches("prez@whitehouse.") must beFalse

  val matcher: RegexMatcher
}
